src/scripts/components/main/data-viewer/data-viewer.tsx

```diff
@@ -150,37 +153,46 @@ const DataViewer: FunctionComponent<Props> = ({
 
   return (
     <div className={styles.dataViewer}>
-      {[mainLayerDetails, compareLayerDetails]
-        .filter((layer): layer is Layer => Boolean(layer))
-        .map(
-          (
-            {id, maxValue, minValue, units, basemap, legendValues, hideLegend},
-            index
-          ) => {
-            if (hideLegend) {
-              return null;
+      {legend &&
+        [mainLayerDetails, compareLayerDetails]
+          .filter((layer): layer is Layer => Boolean(layer))
+          .map(
```

Comment:
A bit ugly here with the destructuring :) Perhaps without it it's cleaner, or extract in a function.